Assets/NewNetworkManager.cs

```diff
@@ -0,0 +1,284 @@
+
+            player1.GetComponent<PlayerBehaviour>().SetOtherPlayer(player2);
+            player2.GetComponent<PlayerBehaviour>().SetOtherPlayer(player1);
+        }
+    }
+
+    /// <summary>
+    /// Called on the server when a client removes a player.
+    /// <para>The default implementation of this function destroys the corresponding player object.</para>
+    /// </summary>
+    /// <param name="conn">The connection to remove the player from.</param>
+    /// <param name="player">The player identity to remove.</param>
+    public override void OnServerRemovePlayer(NetworkConnection conn, NetworkIdentity player)
+    {
+        base.OnServerRemovePlayer(conn, player);
+    }
+
+    /// <summary>
+    /// Called on the server when a network error occurs for a client connection.
+    /// </summary>
+    /// <param name="conn">Connection from client.</param>
+    /// <param name="errorCode">Error code.</param>
+    public override void OnServerError(NetworkConnection conn, int errorCode) { }
+
+    /// <summary>
+    /// Called from ServerChangeScene immediately before SceneManager.LoadSceneAsync is executed
+    /// <para>This allows server to do work / cleanup / prep before the scene changes.</para>
+    /// </summary>
+    /// <param name="newSceneName">Name of the scene that's about to be loaded</param>
+    public override void OnServerChangeScene(string newSceneName) { }
+
+    /// <summary>
+    /// Called on the server when a scene is completed loaded, when the scene load was initiated by the server with ServerChangeScene().
+    /// </summary>
+    /// <param name="sceneName">The name of the new scene.</param>
+    public override void OnServerSceneChanged(string sceneName) { }
+
+    #endregion
+
+    #region Client System Callbacks
+
+    /// <summary>
+    /// Called on the client when connected to a server.
+    /// <para>The default implementation of this function sets the client as ready and adds a player. Override the function to dictate what happens when the client connects.</para>
+    /// </summary>
+    /// <param name="conn">Connection to the server.</param>
+    public override void OnClientConnect(NetworkConnection conn)
+    {
+        base.OnClientConnect(conn);
+
+        if (player1.GetComponent<PlayerBehaviour>().isLocalPlayer) {
+            player1.GetComponent<PlayerBehaviour>().SetOtherPlayer(player2);
+        }
+        else {
+            player2.GetComponent<PlayerBehaviour>().SetOtherPlayer(player1);
+        }
+        
+    }
+
+    /// <summary>
+    /// Called on clients when disconnected from a server.
+    /// <para>This is called on the client when it disconnects from the server. Override this function to decide what happens when the client disconnects.</para>
+    /// </summary>
+    /// <param name="conn">Connection to the server.</param>
+    public override void OnClientDisconnect(NetworkConnection conn)
+    {
+        base.OnClientDisconnect(conn);
```

Comment:
There should probably be some code here to reset player1 & player2 if a client disconnects for whatever reason